pkg/topology/kademlia/kademlia.go

```diff
@@ -1203,16 +1203,26 @@ func (k *Kad) EachPeerRev(f topology.EachPeerFunc) error {
 // onlyReachable wraps the iterator func to filer-out unreachable peers.
 func (k *Kad) onlyReachable(f topology.EachPeerFunc) topology.EachPeerFunc {
 	return func(addr swarm.Address, po uint8) (bool, bool, error) {
-		if !k.collector.Inspect(addr).IsReachable {
+		if k.collector.Inspect(addr).ReachabilityStatus != im.PeerReachabilityStatusPublic {
 			return false, false, nil
 		}
 		return f(addr, po)
 	}
 }
 
 // SetPeerReachability sets the peer reachability status.
-func (k *Kad) SetPeerReachability(addr swarm.Address, isReachable bool) {
-	k.collector.Record(addr, im.PeerReachability(isReachable))
+func (k *Kad) SetPeerReachability(addr swarm.Address, status string) error {
+	rs, err := im.ParseReachabilityStatus(status)
+	if err != nil {
+		return err
+	}
+	k.collector.Record(addr, im.PeerReachability(rs))
+	return nil
+}
+
+// UpdateReachability updates our own reachability status.
+func (k *Kad) UpdateReachability(status string) error {
+	return k.SetPeerReachability(k.base, status)
```

Comment:
this is also misleading with the `Peer` in the naming convention. Since you're setting the node's reachability status and not other peer's reachability status I would suggest to just keep this as a field on kademlia service, and then also expose this information through the topology endpoint